go/libraries/doltcore/sqle/dfunctions/dolt_conflicts_resolve.go

```diff
@@ -0,0 +1,493 @@
+	"github.com/dolthub/dolt/go/store/prolly/tree"
+	"github.com/dolthub/dolt/go/store/types"
+	"github.com/dolthub/dolt/go/store/val"
+)
+
+var ErrConfSchIncompatible = errors.New("the conflict schema's columns are not equal to the current schema's columns, please resolve manually")
+
+// DoltConflictsCatFunc runs a `dolt commit` in the SQL context, committing staged changes to head.
+// Deprecated: please use the version in the dprocedures package
+type DoltConflictsCatFunc struct {
+	children []sql.Expression
+}
+
+// NewDoltConflictsResolveFunc creates a new DoltCommitFunc expression whose children represents the args passed in DOLT_CONFLICTS_RESOLVE.
+// Deprecated: please use the version in the dprocedures package
+func NewDoltConflictsResolveFunc(args ...sql.Expression) (sql.Expression, error) {
+	return &DoltConflictsCatFunc{children: args}, nil
+}
+
+func (d DoltConflictsCatFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	args, err := getDoltArgs(ctx, row, d.Children())
+	if err != nil {
+		return 1, err
+	}
+	return DoDoltConflictsResolve(ctx, args)
+}
+
+func getProllyRowMaps(ctx *sql.Context, vrw types.ValueReadWriter, ns tree.NodeStore, hash hash.Hash, tblName string) (prolly.Map, error) {
+	rootVal, err := doltdb.LoadRootValueFromRootIshAddr(ctx, vrw, ns, hash)
+	tbl, ok, err := rootVal.GetTable(ctx, tblName)
+	if err != nil {
+		return prolly.Map{}, err
+	}
+	if !ok {
+		return prolly.Map{}, doltdb.ErrTableNotFound
+	}
+
+	idx, err := tbl.GetRowData(ctx)
+	if err != nil {
+		return prolly.Map{}, err
+	}
+
+	return durable.ProllyMapFromIndex(idx), nil
+}
+
+func resolveProllyConflicts(ctx *sql.Context, tbl *doltdb.Table, tblName string, sch schema.Schema, ours bool) (*doltdb.Table, error) {
+	var err error
+	artifactIdx, err := tbl.GetArtifacts(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	artifactMap := durable.ProllyMapFromArtifactIndex(artifactIdx)
+	iter, err := artifactMap.IterAllConflicts(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// just get first conflict artifact
+	cnfArt, err := iter.Next(ctx)
+	if err != nil {
+		if err == io.EOF {
+			panic("no conflicts, should be impossible")
+		}
+		return nil, err
+	}
+
+	ourIdx, err := tbl.GetRowData(ctx)
+	if err != nil {
+		return nil, err
+	}
+	ourMap := durable.ProllyMapFromIndex(ourIdx)
+
+	baseRootVal, err := doltdb.LoadRootValueFromRootIshAddr(ctx, tbl.ValueReadWriter(), tbl.NodeStore(), cnfArt.Metadata.BaseRootIsh)
+	baseTbl, ok, err := baseRootVal.GetTable(ctx, tblName)
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		return nil, doltdb.ErrTableNotFound
+	}
+	baseIdx, err := baseTbl.GetRowData(ctx)
+	if err != nil {
+		return nil, err
+	}
+	baseMap := durable.ProllyMapFromIndex(baseIdx)
+
+	theirMap, err := getProllyRowMaps(ctx, tbl.ValueReadWriter(), tbl.NodeStore(), cnfArt.TheirRootIsh, tblName)
+	if err != nil {
+		return nil, err
+	}
+
+	// swap the maps when resolving with theirs
+	if !ours {
+		ourMap, theirMap = theirMap, ourMap
```

Comment:
dolt conflicts resolve --ours on the cli just deletes all the conflicts. When a conflict occurs, the ours value is retained in the working set so no work needs to be done. 

I would check if `ours` is true earlier in this function our prior to it and just delete the conflicts to make that behavior explicit.

Reply:
See here:
https://github.com/dolthub/dolt/blob/7ba66fdc762d0274e4d74bb02dd8d6e7148df247/go/cmd/dolt/commands/cnfcmds/auto_resolve.go#L192-L192